### PR TITLE
DTOs and parsing tests for Bandits

### PR DESCRIPTION
**Pull request by typotter**
- Retyping interface to use generic system interfaces
- DTOs for Bandit models, flags, and evaluation/results.

### CODE REVIEW

File: dot-net-sdk/EppoClient.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ public string GetStringAssignment(string flagKey, string subjectKey, Subject sub
         var result = RuleValidator.EvaluateFlag(configuration, subjectKey, subjectAttributes);
         if (result == null)
         {
-            Logger.Info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
+            Logger.Info("[Eppo SDK] No assigned variation. The idictionary<string, object> attributes did not match any targeting rules");
```
Comment (aarsilv) [inline dot-net-sdk/EppoClient.cs:101]: I don't think you mean to replace this one

File: dot-net-sdk/dto/Subject.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
+using System.Security;
```
Comment (aarsilv) [inline dot-net-sdk/dto/Subject.cs:1]: what is this for?

File: dot-net-sdk/dto/bandit/BanditEvaluation.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace eppo_sdk.dto.bandit;
+
+public record BanditEvaluation{
+
+    [SetsRequiredMembers]
+    public BanditEvaluation(string flagKey,
```
Comment (aarsilv) [inline dot-net-sdk/dto/bandit/BanditEvaluation.cs:8]: Note we also decided to track and log the optimality gap -- the difference between highest weighed action and selected action.

See [python example](https://github.com/Eppo-exp/python-sdk/blob/main/eppo_client/bandit.py#L91))

I know it's a pain to chase a moving target 😛

**Review by aarsilv — APPROVED** [review]
It's coming together! 🎉 
Approving with a few comments I'd love for you to look at at.

File: dot-net-sdk/dto/Subject.cs (hunk 2 above)
Comment (typotter) [reply dot-net-sdk/dto/Subject.cs]: removed

File: dot-net-sdk/dto/bandit/BanditEvaluation.cs (hunk 3 above)
Comment (typotter) [reply dot-net-sdk/dto/bandit/BanditEvaluation.cs:8]: no worries!

**Comment by typotter** [comment]
Thanks for the review, @aarsilv. Made some changes to accommodate bools as well.

**Pull request merged**